pkg/webhook/propagationpolicy/validating.go

```diff
@@ -46,6 +46,13 @@ func (v *ValidatingAdmission) Handle(_ context.Context, req admission.Request) a
 			return admission.Denied(err.Error())
 		}
 	}
+	if req.Operation == admissionv1.Create {
+ 		if policy.Spec.Association {
+			err = fmt.Errorf("spec.association is deprecated, use PropagateDeps instead.")
```

Comment:
```suggestion
			err = fmt.Errorf("spec.association has been deprecated, use PropagateDeps instead.")
```
By the way, we might need to apply this restriction to `ClusterPropagationPolicy` as well.